app/frontend/helpers/ui_helper.js

```diff
@@ -216,6 +216,66 @@ class UiHelper {
     var textLoadingIndicator = this.getCurrentTextLoadingIndicator(tabIndex);
     textLoadingIndicator.hide();
   }
+
+  getFirstVisibleVerseAnchor() {
+    var firstVisibleVerseAnchor = null;
+    var verseListFrame = app_controller.getCurrentVerseListFrame();
+
+    if (verseListFrame != null) {
+      var verseListFrameRect = verseListFrame[0].getBoundingClientRect();
+
+      var currentNavigationPane = app_controller.navigation_pane.getCurrentNavigationPane()[0];
+      var currentNavigationPaneWidth = currentNavigationPane.offsetWidth;
+
+      var firstElementOffsetX = verseListFrameRect.x + currentNavigationPaneWidth + 15;
+      var firstElementOffsetY = verseListFrameRect.y + 15;
+      
+      var firstElement = document.elementFromPoint(firstElementOffsetX, firstElementOffsetY);
+      var currentElement = firstElement;
+
+      if (currentElement.classList != null && currentElement.classList.contains('verse-list')) {
+        // If the current element is the verse-list then we try once more 10 pixels lower.
+        currentElement = document.elementFromPoint(firstElementOffsetX, firstElementOffsetY + 10);
+      }
+
+      if (currentElement.classList != null && 
+          (currentElement.classList.contains('sword-section-title') ||
+           currentElement.classList.contains('tag-browser-verselist-book-header'))) {
+        // We are dealing with a section header element (either sword-section-title or tag-browser-verselist-book-header)
+
+        if (currentElement.previousElementSibling != null &&
+            currentElement.previousElementSibling.nodeName == 'A') {
+
+          currentElement = currentElement.previousElementSibling;
+        }
+      } else {
+        // We are dealing with an element inside a verse-box
+        const MAX_ELEMENT_NEXTING = 7;
+
+        // Traverse up the DOM to find the verse-box
+        for (var i = 0; i < MAX_ELEMENT_NEXTING; i++) {
```

Comment:
Maybe use `let i = 0;` for it to be in more "local" block scope

Reply:
Good point. I guess I should consider this for the whole code base ... you'll find this pattern with `for (var i = 0; ...` all over the place.

Reply:
This is now fixed in the latest commit.